kubeadm cleanup: master -> control-plane (cont.2)

vanduc95 opened the pull request:
This PR continuation of [#73987](https://github.com/kubernetes/kubernetes/pull/73987)

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/assign @neolit123

vanduc95 committed:
kubeadm cleanup: master -> control-plane (cont.2)

neolit123 commented on cmd/kubeadm/app/phases/upgrade/postupgrade.go:
```diff
@@ -207,9 +207,9 @@ func writeKubeletConfigFiles(client clientset.Interface, cfg *kubeadmapi.InitCon
-		// as we handle that ourselves in the markmaster phase
-		// TODO: Maybe we want to do that some time in the future, in order to remove some logic from the markmaster phase?
+		// Write env file with flags for the kubelet to use. We do not need to write the --register-with-taints for the control-plane,
+		// as we handle that ourselves in the markcontrolplane phase
```
`mark-control-plane`

neolit123 commented on cmd/kubeadm/app/phases/upgrade/postupgrade.go:
```diff
@@ -207,9 +207,9 @@ func writeKubeletConfigFiles(client clientset.Interface, cfg *kubeadmapi.InitCon
-		// TODO: Maybe we want to do that some time in the future, in order to remove some logic from the markmaster phase?
+		// Write env file with flags for the kubelet to use. We do not need to write the --register-with-taints for the control-plane,
+		// as we handle that ourselves in the markcontrolplane phase
+		// TODO: Maybe we want to do that some time in the future, in order to remove some logic from the markcontrolplane phase?
```
`mark-control-plane`

neolit123 commented on cmd/kubeadm/app/util/config/cluster.go:
```diff
@@ -59,7 +59,7 @@ func FetchInitConfigurationFromCluster(client clientset.Interface, w io.Writer,
 // getInitConfigurationFromCluster is separate only for testing purposes, don't call it directly, use FetchInitConfigurationFromCluster instead
 func getInitConfigurationFromCluster(kubeconfigDir string, client clientset.Interface, newControlPlane bool) (*kubeadmapi.InitConfiguration, error) {
-	// TODO: This code should support reading the MasterConfiguration key as well for backwards-compat
+	// TODO: This code should support reading the ClusterConfiguration key as well for backwards-compat
```
please remove this comment.
the `MasterConfiguration` key no longer exists in our code base.

neolit123 reviewed (commented):
thanks @vanduc95

neolit123 commented:
/kind cleanup
/priority backlog
/ok-to-test

yagonobre reviewed (commented):
Thanks @vanduc95 
/lgtm

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -143,7 +143,7 @@ const (
 	// the TLS bootstrap to get itself an unique credential
```
vanduc95 there are a lot more Master entries in this file.
we need to change code as well not only comments!

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -191,7 +191,7 @@ const (
-	// LabelNodeRoleMaster specifies that a node is a master
+	// LabelNodeRoleMaster specifies that a node is a control-plane
 	// This is a duplicate definition of the constant in pkg/controller/service/service_controller.go
 	LabelNodeRoleMaster = "node-role.kubernetes.io/master"
```
this is an exception keep it like `LabelNodeRoleMaster`.

neolit123 replied:
@vanduc95 
should have deleted the above comment next to deleting the similar one.

also i'm not sure whether we should keep this or change to `LabelNodeRoleMaster` at this point.
mostly leaning towards change.

but let's wait for more comments...

@rosti @kubernetes/sig-cluster-lifecycle-pr-reviews

rosti replied:
We need to keep it like `LabelNodeRoleMaster`. There is currently a constant named in the same way and used for the same purpose in `pkg/controller/service/service_controller.go` and the idea is at some point to move that constant to a common place (where we won't import a huge package just for it).
Until then, it's best to keep the constant names in sync.

neolit123 replied:
ok, keeping this like so.

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -350,18 +350,18 @@ const (
 
-	// MasterNumCPU is the number of CPUs required on master
+	// MasterNumCPU is the number of CPUs required on control-plane
 	MasterNumCPU = 2
```
`ControlPlaneNumCPU`

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -373,7 +373,7 @@ var (
 
-	// MasterComponents defines the master component names
+	// MasterComponents defines the control-plane component names
 	MasterComponents = []string{KubeAPIServer, KubeControllerManager, KubeScheduler}
```
`ControlPlaneComponents`

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -350,18 +350,18 @@ const (
 var (
-	// MasterTaint is the taint to apply on the PodSpec for being able to run that Pod on the master
+	// MasterTaint is the taint to apply on the PodSpec for being able to run that Pod on the control-plane
 	MasterTaint = v1.Taint{
```
[1]

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -350,18 +350,18 @@ const (
 
-	// MasterToleration is the toleration to apply on the PodSpec for being able to run that Pod on the master
+	// MasterToleration is the toleration to apply on the PodSpec for being able to run that Pod on the control-plane
 	MasterToleration = v1.Toleration{
```
[1]

neolit123 commented on cmd/kubeadm/app/util/endpoint.go:
```diff
@@ -45,7 +45,7 @@ func GetMasterEndpoint(controlPlaneEndpoint string, localEndpoint *kubeadmapi.AP
 
-	// set the master url using localEndpoint.AdvertiseAddress + the localEndpoint.BindPort
+	// set the control-plane url using localEndpoint.AdvertiseAddress + the localEndpoint.BindPort
 	masterURL := &url.URL{
```
`controlPlaneURL`

neolit123 commented on cmd/kubeadm/app/util/endpoint.go:
```diff
@@ -69,7 +69,7 @@ func GetMasterEndpoint(controlPlaneEndpoint string, localEndpoint *kubeadmapi.AP
 
-		// overrides the master url using the controlPlaneAddress (and eventually the bindport)
+		// overrides the control-plane url using the controlPlaneAddress (and eventually the bindport)
 		masterURL = &url.URL{
```
`controlPlaneURL`

neolit123 commented:
@vanduc95 ideally we should modify variables, constants, Function names too.
not only comments.

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -157,9 +157,9 @@ const (
+	// ControlPlaneGroup defines the well-known group for the apiservers. This group is also superuser by default
 	// (i.e. bound to the cluster-admin ClusterRole)
-	MastersGroup = "system:masters"
+	ControlPlaneGroup = "system:masters"
```
please keep `MastersGroup` and `LabelNodeRoleMaster` because they contain "master" as a value.
[1]

neolit123 replied:
actually....let's see what others think and if we can change these too.

rosti replied:
This must be changed to `SystemPrivilegedGroup`

neolit123 commented on cmd/kubeadm/app/constants/constants.go:
```diff
@@ -191,9 +191,9 @@ const (
+	// LabelNodeRoleControlPlane specifies that a node is a control-plane
 	// This is a duplicate definition of the constant in pkg/controller/service/service_controller.go
-	LabelNodeRoleMaster = "node-role.kubernetes.io/master"
+	LabelNodeRoleControlPlane = "node-role.kubernetes.io/master"
```
[1]

rosti replied:
Don't change this one. It's a local copy of an external constant, that will eventually be removed and pointed again to the external reference.

neolit123 reviewed (commented):
thanks for the update @vanduc95 

asking the others:
@kubernetes/sig-cluster-lifecycle-pr-reviews 

how do we feel about renaming `MastersGroup` -> `ControlPlaneGroup` 
`LabelNodeRoleMaster` -> `LabelNodeRoleControlPlane` even if they contain "master" in the string?
SGTM.

neolit123 commented:
@vanduc95 
it's a holiday in the USA so let's wait until tomorrow.

atoato88 reviewed (commented):
If others are in favor of [this](https://github.com/kubernetes/kubernetes/pull/74064#pullrequestreview-204568310), please take into account my review comments.

vanduc95 commented:
Thanks for your review @atoato88.

rosti reviewed (changes requested):
Thanks @vanduc95 !
We have to be careful with some constants though. They were copied locally to avoid importing large external packages and they must retain their old names as they are used throughout k8s sources.

vanduc95 commented:
Thanks @rosti !
This PR has been updated.

rosti reviewed (approved):
Thanks @vanduc95 !
/lgtm

neolit123 reviewed (approved):
@vanduc95 
please rebase to fix merge conflicts.
/approve

vanduc95 commented:
Thanks @neolit123, I fixed merge conflicts.

neolit123 commented on cmd/kubeadm/app/preflight/checks.go:
```diff
@@ -871,7 +871,7 @@ func (ncc NumCPUCheck) Check() (warnings, errorList []error) {
 
-// RunInitMasterChecks executes all individual, applicable to Master node checks.
+// RunInitMasterChecks executes all individual, applicable to master node checks.
 func RunInitMasterChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfiguration, ignorePreflightErrors sets.String) error {
```
looks like we missed this one...
could you please change it to?
`RunInitMasterChecks` -> `RunInitNodeChecks

vanduc95 replied:
yes @neolit123,
I just modified variables, constants and function names under kubeadm/app/constants package. In addition, I also modified in the files which have reference to constants in kubeadm/app/constants.
I will change other package in next PR. Thanks

neolit123 replied:
ok, sounds good.
thanks for that!

/lgtm

vanduc95 replied:
Ok @neolit123, I changed :)

vanduc95 replied:
lol, I just updated PR :D

neolit123 replied:
:+1: let's wait to see if CI passes and will LGTM again.

neolit123 commented on cmd/kubeadm/app/preflight/checks.go:
```diff
@@ -871,7 +871,7 @@ func (ncc NumCPUCheck) Check() (warnings, errorList []error) {
 }
 
-// RunInitMasterChecks executes all individual, applicable to Master node checks.
+// RunInitMasterChecks executes all individual, applicable to master node checks.
```
`master` -> `control-plane`.

vanduc95 replied:
Done

vanduc95 commented:
yes @neolit123, 
I just modified variables, constants and function names under kubeadm/app/constants. In addition, I also modified in the files which have reference to constants in kubeadm/app/constants. 
I will change other package in next PR. Thanks

vanduc95 commented:
/test pull-kubernetes-e2e-gce

neolit123 reviewed (commented):
thanks @vanduc95 

/lgtm
/approve

neolit123 commented:
/retest

neolit123 commented:
/retest
